taichi/codegen/codegen_llvm.cpp

```diff
@@ -407,13 +440,16 @@ void CodeGenLLVM::visit(UnaryOpStmt *stmt) {
   }
   UNARY_INTRINSIC(floor)
   UNARY_INTRINSIC(ceil)
-  else emit_extra_unary(stmt);
+  else {
+    emit_extra_unary(stmt);
+  }
 #undef UNARY_INTRINSIC
 }
 
 void CodeGenLLVM::visit(BinaryOpStmt *stmt) {
```

Comment:
Should we add a helper function like `CreateBinaryOp(BinaryOpType op, bool is_real)` to get rid of these tons of if/else? And seems this whole file is full of if/else...

Reply:
Yup agreed! We should definitely refactor this, it indeed created some trouble with all the branches when working on this

Reply:
I have opened issue #3382 for this. Thanks!